### PR TITLE
Feature/portfoliopage

### PR DESCRIPTION
**Pull request by cristhianbenitez**
Modal Component created

### CODE REVIEW

File: src/components/Modal/Modal.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  StyledDate,
+  StyledCurrency,
+  StyledButton
+} from './Modal.styles';
+
+export class Modal extends Component {
+  constructor(props) {
+    super(props);
+    this.ref = React.createRef(null);
+  }
```
Comment (yoloOnTheBattlefield) [inline src/components/Modal/Modal.js:32]: ref = React.createRef(null); same as you do with the state, no need for a contructor

File: src/components/Modal/Modal.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+                            name="coin"
+                            label="Select Coins"
+                          />
+                        )}
+                        onChange={(_, value) => {
+                          this.setState({
+                            ...this.state,
+                            coin: value?.id || ''
+                          });
+                        }}
```
Comment (yoloOnTheBattlefield) [inline src/components/Modal/Modal.js:130]: please create a class method, don't have this setstate method run in render.

this arrow function gets created everytime render is called, is a waste of resources

File: src/components/Modal/Modal.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+                        label="Purchased Amount"
+                        variant="outlined"
+                        value={this.state.purchasedAmount}
+                        onValueChange={({ value }) =>
+                          this.setState({
+                            ...this.state,
+                            purchasedAmount: value
+                          })
+                        }
```
Comment (yoloOnTheBattlefield) [inline src/components/Modal/Modal.js:153]: same here

File: src/components/Modal/Modal.js [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+                        type="date"
+                        InputLabelProps={{
+                          shrink: true
+                        }}
+                        onChange={(e) => {
+                          this.setState({
+                            ...this.state,
+                            date: e.target.value
+                          });
+                        }}
```
Comment (yoloOnTheBattlefield) [inline src/components/Modal/Modal.js:166]: same here

File: src/components/Modal/Modal.js [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+  handleDateChange = (e) => {
+    this.setState({
+      ...this.state,
+      date: e.target.value
+    });
+  };
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    console.log(this.state);
```
Comment (yoloOnTheBattlefield) [inline src/components/Modal/Modal.js:80]: no console

File: src/pages/Portfolio/Portfolio.js [hunk 6]
```diff
@@ -1,11 +1,40 @@
...
+    this.setState({ ...this.state, isLoading: true });
+    const { data } = await coinGecko.get(`/coins/markets`, {
+      params: {
+        vs_currency: 'usd',
+        order: 'market_cap_desc',
+        per_page: '250'
+      }
+    });
+    this.setState({
+      ...this.state,
```
Comment (yoloOnTheBattlefield) [inline src/pages/Portfolio/Portfolio.js:22]: no need to spread state in here

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: src/components/Modal/Modal.js [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+
+  componentWillUnmount() {
+    document.removeEventListener('mousedown', this.handleClickOutside);
+  }
+  render() {
+    const currentCurrency = localStorage.selection;
+
+    const coinsDetails = this.props.coins.map((coin) => {
+      return { name: coin.name, id: coin.id };
+    });
```
Comment (yoloOnTheBattlefield) [inline src/components/Modal/Modal.js:90]: const coinsDetails = this.props.coins.map(({name, id}) => ({ name, id }));

File: src/components/Modal/Modal.js [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+                    <LeftContent>
+                      <CoinImageContainer>
+                        <CoinImage
+                          src={minimizedImage}
+                          alt={coinInformation[0]?.id}
+                        />
+                      </CoinImageContainer>
+                      <CoinNameText>
+                        {coinInformation[0]?.name}
+                        {`(${coinInformation[0]?.symbol.toUpperCase()})`}
```
Comment (yoloOnTheBattlefield) [inline src/components/Modal/Modal.js:119]: you are using coinInformation[0] many times, extract it in a variable

File: src/components/ModalAutocomplete/ModalAutocomplete.js [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+} from './ModalAutocomplete.styles';
+
+export class ModalAutocomplete extends Component {
+  state = {
+    isDropdownVisible: false,
+    text: '',
+    suggestions: [...this.props.data]
+  };
+
+  onTextChanged = (e) => {
```
Comment (yoloOnTheBattlefield) [inline src/components/ModalAutocomplete/ModalAutocomplete.js:21]: onTextChange*

File: src/pages/Portfolio/Portfolio.js [hunk 10]
```diff
@@ -1,11 +1,39 @@
...
+    isLoading: false,
+    coins: []
+  };
+  getCoins = async () => {
+    this.setState({ ...this.state, isLoading: true });
+    const { data } = await coinGecko.get(`/coins/markets`, {
+      params: {
+        vs_currency: 'usd',
+        order: 'market_cap_desc',
+        per_page: '250'
```
Comment (yoloOnTheBattlefield) [inline src/pages/Portfolio/Portfolio.js:18]: use camelCase

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: src/pages/Portfolio/Portfolio.js (hunk 10 above)
Comment (cristhianbenitez) [reply src/pages/Portfolio/Portfolio.js:18]: how can I use camel case for these params from the API ???

**Pull request merged**